src/tools/tests/kotlin-codegen-test-suite.ts

```diff
@@ -0,0 +1,110 @@
+/**
+ * @license
+ * Copyright (c) 2019 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {SchemaGraph} from '../schema2graph.js';
+import {generateSchema} from '../kotlin-schema-generator.js';
+import {ManifestCodegenUnitTest} from '../codegen-unit-test.js';
+import {KTExtracter} from '../kotlin-refinement-generator.js';
+import {CodegenUnitTest} from '../codegen-unit-test.js';
+import {Manifest} from '../../runtime/manifest.js';
+import {Flags} from '../../runtime/flags.js';
+import {KotlinEntityGenerator} from '../kotlin-entity-generator.js';
+import {Schema2Kotlin} from '../schema2kotlin.js';
+import {generateConnectionSpecType, generateType} from '../kotlin-type-generator.js';
+
+/**
+ * A Suite of unit tests for Kotlin Codegen.
+ * The test data is inside .cgtest files.
+ */
+export const testSuite: CodegenUnitTest[] = [
+  new class extends ManifestCodegenUnitTest {
+    constructor() {
+      super(
+        'Kotlin Schema Generation',
+        'kotlin-schema-generation.cgtest'
```

Comment:
Is this a correct understanding? -- Here, we register what we generate per file. This way, a particle spec input will generate a type in one file, whereas it will generate a schema in another file. 

If this is the case, we should document how to add a new class of tests / generations. Or, is it possible to specify what's being generated at the top of the cgtest files? 🤔